src/odb_backend.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2010-2019 The pygit2 contributors
+ *
+ * This file is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License, version 2,
+ * as published by the Free Software Foundation.
+ *
+ * In addition to the permissions in the GNU General Public License,
+ * the authors give you unlimited permission to link the compiled
+ * version of this file into combinations with other programs,
+ * and to distribute those combinations without any restriction
+ * coming from the use of this file.  (The General Public License
+ * restrictions do apply in other respects; for example, they cover
+ * modification of the file, and distribution when not linked into
+ * a combined executable.)
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+#include "error.h"
+#include "object.h"
+#include "oid.h"
+#include "types.h"
+#include "utils.h"
+#include <git2/odb_backend.h>
+#include <git2/sys/alloc.h>
+#include <git2/sys/odb_backend.h>
+
+
+PyDoc_STRVAR(OdbBackend_read__doc__,
+    "read(oid) -> (type, data, size)\n"
+    "\n"
+    "Read raw object data from this odb backend.\n");
+
+PyObject *
+OdbBackend_read(OdbBackend *self, PyObject *py_hex)
```

Comment:
Note, in keeping with the libgit2 API design, I have no intention of refactoring `Odb_read` to go through this code for the sake of DRY.